### PR TITLE
Add condition to check dependency MeshSDFilter folder

### PR DESCRIPTION
**Pull request by JackBoosY**
## Implementation remarks

FTR https://github.com/alicevision/AliceVision/issues/767#issuecomment-2042771385

The check should requires `ALICEVISION_USE_MESHSDFILTER` is `ON`.

### CODE REVIEW

**Comment by servantftransperfect** [comment]
Sorry, but this PR is not valid or at least incomplete. 

I think this condition is meant to check that the submodule*s* have been retrieved and exit early. There is not only MeshSDFilter but also NanoFlann. If we merge this PR, then we should add a second test for nanoflann submodule.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -178,7 +178,7 @@ endmacro(add_target_properties)
 # ==============================================================================
 # Check that submodule have been initialized and updated
 # ==============================================================================
-if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/MeshSDFilter/CMakeLists.txt)
+if(ALICEVISION_USE_MESHSDFILTER AND NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/MeshSDFilter/CMakeLists.txt)
```
Comment (JackBoosY) [inline src/CMakeLists.txt:181]: > Sorry, but this PR is not valid or at least incomplete.
> 
> I think this condition is meant to check that the submodule_s_ have been retrieved and exit early. There is not only MeshSDFilter but also NanoFlann. If we merge this PR, then we should add a second test for nanoflann submodule.

But this is only for `MeshSDFilter`, right?

**Comment by simogasp** [comment]
> Sorry, but this PR is not valid or at least incomplete.
> 
> I think this condition is meant to check that the submodule_s_ have been retrieved and exit early. There is not only MeshSDFilter but also NanoFlann. If we merge this PR, then we should add a second test for nanoflann submodule.

That's the point. Nanoflann is mandatory and we don't check it, MeshSDFilter and we check. So either we don't check either of them or we check both of them. MeshSDFilter should be checked the way this PR suggests because it's optional, if anything we should add a check for nanoflann

**Comment by JackBoosY** [comment]
> So either we don't check either of them or we check both of them.

Should I also add check for NanoFlann in this PR?

**Comment by JackBoosY** [comment]
Ping @simogasp for review again.

**Review by simogasp — APPROVED** [review]

**Comment by simogasp** [comment]
@JackBoosY  Thanks!

**Pull request merged**